### ISSUE TITLE
Input textures are not displayed.

### ISSUE THREAD

**Issue by Megaxela** [baldurk/renderdoc#736]
Hello. When trying to capture application with `qrenderdoc` there is no input textures displayed. Even renderdoc result frame texture is differ from application frame.
I'm running linux:
`uname -a` -> `Linux ArchLinuxLaptop 4.12.10-1-ARCH #1 SMP PREEMPT Wed Aug 30 12:18:42 CEST 2017 x86_64 GNU/Linux`

Testing on `v0.x` branch. Branch `v1.x` is not even receiving captures from application. 

Attaching `glewinfo` and `glxinfo` results.
[glewinfo.txt](https://github.com/baldurk/renderdoc/files/1284855/glewinfo.txt)
[glxinfo.txt](https://github.com/baldurk/renderdoc/files/1284854/glxinfo.txt)

**Comment by baldurk** [comment]
Hey, can you give more details about what precisely is going wrong, and any information or material I use to try and reproduce the problem?

Are the textures showing as bound, but with incorrect (or empty) contents? Are the texture showing as unbound? Or are the inputs missing completely as if the shader doesn't use any textures at all?

Can you share a capture, or the program itself? Also pasting the output log (from Help -> View Diagnostic Log) is useful to see if there are any error messages there.

**Comment by Megaxela** [comment]
Textures are displayed as bounded but their content is empty. Sometimes them contains strange junk.
This [archive](https://github.com/baldurk/renderdoc/files/1285688/TEST.tar.gz) contains test executable, this executable capture result and diagnostic log. 
Executable `ldd` result:
```
        linux-vdso.so.1 (0x00007ffe19dd5000)
        libSOIL.so.1 => /usr/lib/libSOIL.so.1 (0x00007f444479d000)
        libfreetype.so.6 => /usr/lib/libfreetype.so.6 (0x00007f44444d6000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007f44442bf000)
        libGL.so.1 => /usr/lib/libGL.so.1 (0x00007f4444034000)
        libX11.so.6 => /usr/lib/libX11.so.6 (0x00007f4443cf5000)
        libXrandr.so.2 => /usr/lib/libXrandr.so.2 (0x00007f4443aea000)
        libXi.so.6 => /usr/lib/libXi.so.6 (0x00007f44438d9000)
        libdl.so.2 => /usr/lib/libdl.so.2 (0x00007f44436d5000)
        libXinerama.so.1 => /usr/lib/libXinerama.so.1 (0x00007f44434d2000)
        libXcursor.so.1 => /usr/lib/libXcursor.so.1 (0x00007f44432c7000)
        libXxf86vm.so.1 => /usr/lib/libXxf86vm.so.1 (0x00007f44430c1000)
        libGLEW.so.2.1 => /usr/lib/libGLEW.so.2.1 (0x00007f4442e19000)
        libglfw.so.3 => /usr/lib/libglfw.so.3 (0x00007f4442bfb000)
        libSM.so.6 => /usr/lib/libSM.so.6 (0x00007f44429f3000)
        libICE.so.6 => /usr/lib/libICE.so.6 (0x00007f44427d6000)
        libXext.so.6 => /usr/lib/libXext.so.6 (0x00007f44425c4000)
        libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f44423a6000)
        librt.so.1 => /usr/lib/librt.so.1 (0x00007f444219e000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f4441e16000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007f4441b04000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f44418ed000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007f4441547000)
        libbz2.so.1.0 => /usr/lib/libbz2.so.1.0 (0x00007f4441337000)
        libpng16.so.16 => /usr/lib/libpng16.so.16 (0x00007f4441100000)
        libharfbuzz.so.0 => /usr/lib/libharfbuzz.so.0 (0x00007f4440e66000)
        libGLX.so.0 => /usr/lib/libGLX.so.0 (0x00007f4440c34000)
        libGLdispatch.so.0 => /usr/lib/libGLdispatch.so.0 (0x00007f444097e000)
        libxcb.so.1 => /usr/lib/libxcb.so.1 (0x00007f4440755000)
        libXrender.so.1 => /usr/lib/libXrender.so.1 (0x00007f444054a000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f44449b4000)
        libXfixes.so.3 => /usr/lib/libXfixes.so.3 (0x00007f4440344000)
        libuuid.so.1 => /usr/lib/libuuid.so.1 (0x00007f444013f000)
        libglib-2.0.so.0 => /usr/lib/libglib-2.0.so.0 (0x00007f443fe2c000)
        libgraphite2.so.3 => /usr/lib/libgraphite2.so.3 (0x00007f443fc00000)
        libXau.so.6 => /usr/lib/libXau.so.6 (0x00007f443f9fc000)
        libXdmcp.so.6 => /usr/lib/libXdmcp.so.6 (0x00007f443f7f6000)
        libpcre.so.1 => /usr/lib/libpcre.so.1 (0x00007f443f583000)
```

**Comment by baldurk** [comment]
There were a couple of issues here. One thing was the recent change in #734 caused issues on matrix elements being passed between vertices, since they were split into "foo:row0", "foo:row1" etc. Fixing that though and passing just "foo", the program then fails to link for me on intel mesa saying that the variable was undeclared. I think this is just a driver bug - maybe again poor handling of matrix interpolators. So I added a fallback to remove the varyings if the program fails to link, as it's not required but just working around some different driver issues. CC @michaelrgb in case there are unexpected knock-ons from that.

The missing contents of textures also seems like it's maybe a mesa bug. It complained and then failed to do a glCopyImageSubData between a texture created with internal format `GL_RGBA` (unsized) and one created with `GL_RGBA8`. That seems like a bug to me but I imagine it's probably undefined spec behaviour as the unsized formats are kind of a leftover from ancient GL. I added a fix for this that converts unsized formats to sized formats before passing to GL, instead of only doing it for internal handling. I've only put that on v1.x though as I'm not sure how stable it is. I fixed the issue you ran into where v1.x wasn't able to capture, so hopefully now everything should work on the v1.x branch.
